Automattic/wp-calypso
# Jetpack Backup: Add the logic to update backup retention settings #73088

## Proposed Changes

* This implements the logic needed to call WPCOM API to update the backup retention period from a Calypso component.
* Also includes a selector `getBackupRetentionUpdateRequestStatus` to query the update request's status.

## Testing Instructions



* Unit tests:  `yarn run test-client client/state/rewind`
* Using Redux Dev Tool:
  1) Test `retention.updateRequestStatus` state, which is updated while calling the update API.  
  
      Initially, the `state.rewind[BLOG_ID].updateBackupRetentionRequestStatus` should be in the `unsubmitted` state.

      Now, for the following action when dispatched:
      ```javascript
      {
        type: 'JETPACK_BACKUP_RETENTION_UPDATE',
        siteId: '<blog-id>',
        rententionDays: 7,
      }
      ```
      the `state.rewind[BLOG_ID].updateBackupRetentionRequestStatus` should be in the `pending` state.
      
      For the following action:
      ```javascript
      {
        type: 'JETPACK_BACKUP_RETENTION_UPDATE_SUCCESS',
        siteId: '<blog-id>',
      }
      ```
      the `state.rewind[BLOG_ID].updateBackupRetentionRequestStatus` should be in the `success` state.
    
      For the following action:
      ```javascript
      {
        type: 'JETPACK_BACKUP_RETENTION_UPDATE_ERROR',
        siteId: '<blog-id>',
      }
      ```
      the `state.rewind[BLOG_ID].updateBackupRetentionRequestStatus` should be in the `failed` state.



## Pre-merge Checklist
